llvm/lib/Target/AIE/AIE2AddrSpace.h

```diff
@@ -40,6 +42,71 @@ enum class AddressSpaces {
 
 enum class AIEBanks { A, B, C, D, TileMemory };
 
```

Comment:
I was thinking if we could create and AIEBaseAddrSpace (class), and then a derivation like AIE2AddrSpace.h (subclass) to prametrize  AIEBaseAAWrapperPass. I think your AA implementation is AIE version independent (if you provide an AIEBaseAddrSpace object).